src/main/java/org/prebid/server/bidder/mgid/MgidBidder.java

```diff
@@ -0,0 +1,217 @@
+    }
+
+    private List<HttpRequest<BidRequest>> makeBatchRequest(BidRequest bidRequest,
+                                                           List<ImpWithExt<ExtImpMgid>> impsWithExts) {
+        return splitIntoMapOfImpByEndpoint(impsWithExts).entrySet().stream()
+                .map(impWithExtByEndpoint -> makeRequest(impWithExtByEndpoint.getKey(), bidRequest,
+                        impWithExtByEndpoint.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    private Map<String, List<Imp>> splitIntoMapOfImpByEndpoint(
+            List<ImpWithExt<ExtImpMgid>> impsWithExts) {
+        return impsWithExts.stream()
+                .collect(Collectors.groupingBy(this::createEndpointFromAccountId,
+                        Collectors.mapping(ImpWithExt::getImp, Collectors.toList())));
+    }
+
+    private String createEndpointFromAccountId(ImpWithExt<ExtImpMgid> mgidExt) {
+        return mgidEndpoint + mgidExt.getImpExt().getAccountId();
+    }
+
+    private HttpRequest<BidRequest> makeRequest(String endpoint, BidRequest bidRequest,
+                                                List<Imp> imps) {
+        final BidRequest outgoingRequest = bidRequest.toBuilder()
+                .imp(imps)
+                .build();
+
+        final String body = Json.encode(outgoingRequest);
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpoint)
+                .body(body)
+                .headers(BidderUtil.headers())
+                .payload(outgoingRequest)
+                .build();
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall,
+                                                  BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = Json
+                    .decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse),
+                    Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid
+                        .of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), getBidCurrency()))
```

Comment:
Please check Go implementation as further logic is not correct - you've missed extracting their specific extension from bidResp.SeatBid.Bid[...].ext, from which you then should resolve bid type.

At this point you need to also parse the bid.ext in their specific model and use their bid.ext to determine the bid type. 

Once again, don't copy and try to fit the logic - go it step-by-step and use existing bidders as an example.